frontend/src/lib/lemon-ui/LemonInputSelect/LemonInputSelect.tsx

```diff
@@ -0,0 +1,269 @@
+    }, [visibleOptions.length])
+
+    const setInputValue = (newValue: string): void => {
+        _setInputValue(newValue)
+        onInputChange?.(inputValue)
+    }
+
+    const _onActionItem = (item: string): void => {
+        let newValues = [...values]
+        if (values.includes(item)) {
+            // Remove the item
+            if (mode === 'single') {
+                newValues = []
+            } else {
+                newValues.splice(values.indexOf(item), 1)
+            }
+        } else {
+            // Add the item
+            if (mode === 'single') {
+                newValues = [item]
+            } else {
+                newValues.push(item)
+            }
+
+            setInputValue('')
+        }
+
+        onChange?.(newValues)
+    }
+
+    const _onBlur = (): void => {
+        // We need to add a delay as a click could be in the popover or the input wrapper which refocuses
+        setTimeout(() => {
+            if (popoverFocusRef.current) {
+                popoverFocusRef.current = false
+                inputRef.current?.focus()
+                _onFocus()
+                return
+            }
+            if (allowCustomValues && inputValue.trim() && !values.includes(inputValue)) {
+                _onActionItem(inputValue.trim())
+            } else {
+                setInputValue('')
+            }
+            setShowPopover(false)
+        }, 100)
+    }
+
+    const _onFocus = (): void => {
+        setShowPopover(true)
+        popoverFocusRef.current = true
+    }
+
+    const _onKeyDown = (e: React.KeyboardEvent<HTMLInputElement>): void => {
+        if (e.key === 'Enter') {
+            e.preventDefault()
+
+            const itemToAdd = visibleOptions[selectedIndex]?.key
+            if (itemToAdd) {
+                _onActionItem(visibleOptions[selectedIndex]?.key)
+            }
+        } else if (e.key === 'Backspace') {
+            if (!inputValue) {
+                e.preventDefault()
+                const newValues = [...values]
+                newValues.pop()
+                onChange?.(newValues)
+            }
+        } else if (e.key === 'ArrowDown') {
+            e.preventDefault()
+            setSelectedIndex(Math.min(selectedIndex + 1, options.length - 1))
+        } else if (e.key === 'ArrowUp') {
+            e.preventDefault()
+            setSelectedIndex(Math.max(selectedIndex - 1, 0))
+        }
+    }
+
+    // TRICKY: We don't want the popover to affect the snack buttons
+    const prefix = (
+        <PopoverReferenceContext.Provider value={null}>
+            <>
+                {values.map((value) => {
+                    const option = options.find((option) => option.key === value) ?? {
+                        label: value,
+                        labelComponent: null,
+                    }
+                    return (
+                        <>
+                            <LemonSnack title={option?.label} onClose={() => _onActionItem(value)}>
```

Comment:
We should not be showing a `Snack` for a single-value selection. It's like having a checkbox where only single selection is allowed – technically works, but misleads the user initially. I suggest that this case mirrors `LemonSelect`, just with an input during selection.
This experience for example is annoying:
![unexpected](https://github.com/PostHog/posthog/assets/4550621/c343f3fe-6a2c-4640-8100-a532ac6d2f7a)
The UI is oriented around selecting multiple values, it's not expected that the previous selection is replaced wholesale without you clicking "x"

Reply:
Side issue: `LemonSnack` has the text selection cursor, but is in fact unselectable
![selectability](https://github.com/PostHog/posthog/assets/4550621/62ed6700-2384-472d-a4f1-03993bbd1888)

Reply:
My intention with this PR was - remove AntD - not fundamentally change implementation.

Feels like a sensible follow up PR :D

Reply:
The experience is much different in this case though:
![2024-03-18 17 27 19](https://github.com/PostHog/posthog/assets/4550621/bda3fe95-833a-4e64-a87d-4966660990e0)